### ISSUE TITLE
Conan version 1.x on Apple silicon issue : ERROR: Unknown command '-B'

### ISSUE THREAD

**Issue by regalir** [conan-io/conan#15609]
### What is your question?

Hi, I am running Conan on an Apple Silicon to download the ninja package but I keep getting the message:
```
'-B' is not a Conan command. See 'conan --help'.
ERROR: Unknown command '-B'
```
I am thinking that it may be related to the way Conan behaves on an Apple Silicon machine and that -B may not even be an actual command but, maybe, a badly decoded character string? I tried to find anything related to the -B on the web but nothing showed up. I copied some conan functions locally so I could put print statements to debug. But the string seems to be printed out by some subprocess because it always show up after all my traces. Then, I wondered if it was only the ninja package that had this issue so I tried with the magic_enum (just a random package I choosed because it looked small enough for my testing purposes) and, it too, display the same error message so it seems that this is a general thing when trying to download packages from conan-center from a Mac Silicon (Windows and Linux system don't have this "issue" :-(

At this point, my question is: do you have any idea where this message come from and how it can be prevented?

Please, allow me to dig into all the details of my investigations. But, first, the mandatory machine info:

```
Machine: Mac Studio
Chip: Apple M1 Ultra
macOS: Ventura (version 13.6)

Conan version 1.59.0
```
For the testing purposes, I have downloaded the _conanfile.py_ and _conandata.yml_ for the _magic_enum_ and _ninja_ packages from the conan-center and put them in temporary folders (respectively: ./testing/_magicenum_ and ./testing/ninja). 

All my investigations were made for the ninja package so I ended up grabbing the _get_ and _download_ functions original source code and created the _getme_ and _downloadme_ local functions so I could put some traces in the code. I also commented out the code of the _generate_, _build_, _package_ and _package_info_ function and just put a print(">>> ...") statement to simplify the execution. This is the output I am getting:

```
regalir@MTLRKJLYQV6G6 3dsmax-conan-recipes % ~/.local/conan create testing/ninja/conanfile.py 1.11.1@local/testing -pr=maya-2025-mac-release --build missing      
Exporting package recipe
ninja/1.11.1@local/testing exports: File 'conandata.yml' found. Exporting it...
ninja/1.11.1@local/testing exports: Copied 1 '.yml' file: conandata.yml
ninja/1.11.1@local/testing: A new conanfile.py version was exported
ninja/1.11.1@local/testing: Folder: /Users/regalir/.conan/data/ninja/1.11.1/local/testing/export
ninja/1.11.1@local/testing: Using the exported files summary hash as the recipe revision: 1a7c1ae005b79800c0dcea2823ea006e 
ninja/1.11.1@local/testing: Exported revision: 1a7c1ae005b79800c0dcea2823ea006e
Configuration:
[settings]
arch=armv8
arch_build=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=13
os=Macos
os_build=Macos
ninja:build_type=Release
nasm:build_type=Release
cmake:build_type=Release
[options]
[build_requires]
[env]

Installing package: ninja/1.11.1@local/testing
Requirements
    ninja/1.11.1@local/testing from local cache - Cache
Packages
    ninja/1.11.1@local/testing:3eb32b976fb0626f3c40a8b87ace72156aa44204 - Build

Installing (downloading, building) binaries...
ninja/1.11.1@local/testing: Configuring sources in /Users/regalir/.conan/data/ninja/1.11.1/local/testing/source/src
>>> source
>>> getme: url=https://github.com/ninja-build/ninja/archive/v1.11.1.tar.gz
>>> filename=v1.11.1.tar.gz
>>> download
>>> downloadme.pass1
>>> run_downloader
>>>    verify=True
>>>    download_cache=None
>>>    user_download=True
>>>    url=https://github.com/ninja-build/ninja/archive/v1.11.1.tar.gz
>>>    file_path=v1.11.1.tar.gz
>>>    retry=1
>>>    retry_wait=5
>>>    overwrite=True
>>>    auth=None
>>>    headers=None
.>>    md5=None
>>>    sha1=None
>>>    sha256=31747ae633213f1eda3842686f83c2aa1412e0f5691d1c14dbbcc67fe7400cea
Downloading v1.11.1.tar.gz completed [224.10k]                                           ninja/1.11.1@local/testing: l/testing: 
>>> returned from run_downloader
ninja/1.11.1@local/testing: 
>>> return from download._download_file
>>> unzip
>>> unlink
>>> source.done
ninja/1.11.1@local/testing: Copying sources to build folder
'-B' is not a Conan command. See 'conan --help'.

ERROR: Unknown command '-B'
ninja/1.11.1@local/testing: Building your package in /Users/regalir/.conan/data/ninja/1.11.1/local/testing/build/3eb32b976fb0626f3c40a8b87ace72156aa44204
ninja/1.11.1@local/testing: Generator txt created conanbuildinfo.txt
ninja/1.11.1@local/testing: Calling generate()
>>> generate (pass)
ninja/1.11.1@local/testing: Aggregating env generators
ninja/1.11.1@local/testing: Calling build()
>>> build (pass)
ninja/1.11.1@local/testing: Package '3eb32b976fb0626f3c40a8b87ace72156aa44204' built
ninja/1.11.1@local/testing: Build folder /Users/regalir/.conan/data/ninja/1.11.1/local/testing/build/3eb32b976fb0626f3c40a8b87ace72156aa44204/build/Release
ninja/1.11.1@local/testing: Generated conaninfo.txt
ninja/1.11.1@local/testing: Generated conanbuildinfo.txt
ninja/1.11.1@local/testing: Generating the package
ninja/1.11.1@local/testing: Package folder /Users/regalir/.conan/data/ninja/1.11.1/local/testing/package/3eb32b976fb0626f3c40a8b87ace72156aa44204
ninja/1.11.1@local/testing: Calling package()
>>> package (pass)
ninja/1.11.1@local/testing package(): WARN: No files in this package!
ninja/1.11.1@local/testing: Package '3eb32b976fb0626f3c40a8b87ace72156aa44204' created
ninja/1.11.1@local/testing: Created package revision 5f1337551a6b504be6c28c73fc6641cf
>>> pacakage info (pass)

```
As I mentioned earlier, at some point I wondered if it was a "general" thing or only related to the ninja package so I tried with the magic_enum package (plain files, no modifs for traces whatsoever) and when I ran:

`~/.local/conan create testing/_magicenum_/conanfile.py 0.9.0@local/testing `
I  got the following output:
```
regalir@MTLRKJLYQV6G6 3dsmax-conan-recipes % ~/.local/conan create testing/_magicenum_/conanfile.py 0.9.0@local/testing --build=missing
Exporting package recipe
magic_enum/0.9.0@local/testing exports: File 'conandata.yml' found. Exporting it...
magic_enum/0.9.0@local/testing exports: Copied 1 '.yml' file: conandata.yml
magic_enum/0.9.0@local/testing: A new conanfile.py version was exported
magic_enum/0.9.0@local/testing: Folder: /Users/regalir/.conan/data/magic_enum/0.9.0/local/testing/export
magic_enum/0.9.0@local/testing: Using the exported files summary hash as the recipe revision: 6badeab3fa8882d7fdf5c7f7ee756df1 
magic_enum/0.9.0@local/testing: Exported revision: 6badeab3fa8882d7fdf5c7f7ee756df1
Configuration:
[settings]
arch=armv8
arch_build=armv8
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=15
os=Macos
os_build=Macos
[options]
[build_requires]
[env]

Installing package: magic_enum/0.9.0@local/testing
Requirements
    magic_enum/0.9.0@local/testing from local cache - Cache
Packages
    magic_enum/0.9.0@local/testing:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Build

Installing (downloading, building) binaries...
magic_enum/0.9.0@local/testing: Configuring sources in /Users/regalir/.conan/data/magic_enum/0.9.0/local/testing/source/src
Downloading v0.9.0.tar.gz completed [188.77k]                                            magic_enum/0.9.0@local/testing: local/testing: 
magic_enum/0.9.0@local/testing: 
magic_enum/0.9.0@local/testing: Building your package in /Users/regalir/.conan/data/magic_enum/0.9.0/local/testing/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
magic_enum/0.9.0@local/testing: Generator txt created conanbuildinfo.txt
magic_enum/0.9.0@local/testing: Aggregating env generators
magic_enum/0.9.0@local/testing: Calling build()
magic_enum/0.9.0@local/testing: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' built
magic_enum/0.9.0@local/testing: Build folder /Users/regalir/.conan/data/magic_enum/0.9.0/local/testing/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/build-release
magic_enum/0.9.0@local/testing: Generated conaninfo.txt
magic_enum/0.9.0@local/testing: Generated conanbuildinfo.txt
magic_enum/0.9.0@local/testing: Generating the package
magic_enum/0.9.0@local/testing: Package folder /Users/regalir/.conan/data/magic_enum/0.9.0/local/testing/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
magic_enum/0.9.0@local/testing: Calling package()
magic_enum/0.9.0@local/testing: Copied 6 '.hpp' files
magic_enum/0.9.0@local/testing: Copied 1 file: LICENSE
magic_enum/0.9.0@local/testing package(): Packaged 1 file: LICENSE
magic_enum/0.9.0@local/testing package(): Packaged 6 '.hpp' files
magic_enum/0.9.0@local/testing: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' created
magic_enum/0.9.0@local/testing: Created package revision e956c6c354712ca4b9ef73ff05a948ab
regalir@MTLRKJLYQV6G6 3dsmax-conan-recipes % '-B' is not a Conan command. See 'conan --help'.

ERROR: Unknown command '-B'
```

So it "confirm" that the -B message is not related to which package is downloaded. But at this point I don't know what else to try.
Any information, advice, suggestions are welcome.

Thanks

**Comment by Ahajha** [comment]
Total shot in the dark, perhaps it's trying to run Python, but maybe you only have Conan installed as a standalone executable, so it's trying to use the Conan executable as if it was Python? I do see that `-B` is a valid `python` flag.

**Comment by regalir** [comment]
Indeed Conan is installed as a standalone and I call it directly (it is not even inside the Python site-packages folder) but this does not seem to be an issue for a bunch of other packages that are retrieved from a local "repository". During my previous investigation it looked like the issue may happen somewhere inside the downloading of the tar.gz file (or maybe during the extraction?). The difference between the packages "consumed" directly from conan-center and the ones locally is that; locally we use "repo" and "tag" instead of "url" and "sha256". The source() function in the conanfile.py looks like this:
```
    def source(self):
        git = tools.Git(folder=self.source_folder, username=os.getenv("GITHUB_LOGIN_USERNAME"),
                        password=os.getenv("GITHUB_LOGIN_PASSWORD"), )
        git.clone(self.conan_data["sources"][self.version]['repo'],
                  self.conan_data["sources"][self.version]['tag'])
```

instead of the original way:
```
    def source(self):
        get(self, **self.conan_data["sources"][self.version], strip_root=True)
```

Can it be that the "get" and/or some other internal functions/module (most likely the "run_downloader") are trying to avoid generating the .pyc files and, because they are assuming that the module is called from within Python, they call the "executable" (in this case: Conan) with the -B option? If this is the case is there something that must be done on our side? Note that using Conan as a standalone product is a requirement for us.

Thank you!

**Comment by uilianries** [comment]
This issue seems be related to Conan client, not to a specific package in Conan Center. I'll be moving it to the Conan client repository.

**Comment by memsharded** [comment]
Hi @regalir 

thanks for your report.

I am a big surprised about this. If you use the previous ``get(self, **self.conan_data["sources"][self.version], strip_root=True)``, does it still gets the same error, or it works correctly? I am just trying to think where it can come from.

**Comment by memsharded** [comment]
I am not sure if it could be a broken/corrupted Conan installation. It would also be worth trying to re-install Conan, or use Conan installed from ``pip`` instead (inside a Python virtualenv recommended)

**Comment by regalir** [comment]
Hi @memsharded, maybe the following can shed some light!

This morning I've been informed that **we** have built conan from source because we want a self-contained version that is automatically downloaded by our build processes so every developer does not need to care about which version of Pyton & conan need to be installed on their machines!

Knowing this, I have locally downloaded the sources of conan 1.61 to re-build it again (just in case the current package we generated a while ago was "corrupted". I simply followed the standard process:

on a MacOS shell:
```
regalir@MTLRKJLYQV6G6 % git clone https://github.com/conan-io/conan conan_src
regalir@MTLRKJLYQV6G6 % cd conan_src
regalir@MTLRKJLYQV6G6 % git checkout release/1.61
regalir@MTLRKJLYQV6G6 % pip install -e .
regalir@MTLRKJLYQV6G6 % python pyinstaller.py
```
however, before running the pyinstaller, I went inside the file **.../conan_src/conans/client/command.py** and added the trace:
```
def run(self, *args):
        """HIDDEN: entry point for executing commands, dispatcher to class
        methods
        """
        ret_code = SUCCESS
        print(">>> command.run")
        print(args)
        try:
            try:
...
```
so, when executing:
`/Users/regalir/Development/tests/conan_src/pyinstaller/dist/conan/conan create testing/ninja/conanfile.py 1.11.1@local/testing -pr=maya-2025-mac-release --build missing`

I can have some idea of what is going on! And here is the full output:
```
regalir@MTLRKJLYQV6G6 3dsmax-conan-recipes % /Users/regalir/Development/tests/conan_src/pyinstaller/dist/conan/conan create testing/ninja/conanfile.py 1.11.1@local/testing -pr=maya-2025-mac-release --build missing
>>> command.run
(['create', 'testing/ninja/conanfile.py', '1.11.1@local/testing', '-pr=maya-2025-mac-release', '--build', 'missing'],)
Exporting package recipe
ninja/1.11.1@local/testing exports: File 'conandata.yml' found. Exporting it...
ninja/1.11.1@local/testing exports: Copied 1 '.yml' file: conandata.yml
ninja/1.11.1@local/testing: A new conanfile.py version was exported
ninja/1.11.1@local/testing: Folder: /Users/regalir/.conan/data/ninja/1.11.1/local/testing/export
ninja/1.11.1@local/testing: Using the exported files summary hash as the recipe revision: 1a7c1ae005b79800c0dcea2823ea006e 
ninja/1.11.1@local/testing: Exported revision: 1a7c1ae005b79800c0dcea2823ea006e
Configuration:
[settings]
arch=armv8
arch_build=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=13
os=Macos
os_build=Macos
ninja:build_type=Release
nasm:build_type=Release
cmake:build_type=Release
[options]
[build_requires]
[env]

Installing package: ninja/1.11.1@local/testing
Requirements
    ninja/1.11.1@local/testing from local cache - Cache
Packages
    ninja/1.11.1@local/testing:3eb32b976fb0626f3c40a8b87ace72156aa44204 - Build

Installing (downloading, building) binaries...
ninja/1.11.1@local/testing: Configuring sources in /Users/regalir/.conan/data/ninja/1.11.1/local/testing/source/src
Downloading v1.11.1.tar.gz completed [224.10k]                                           ninja/1.11.1@local/testing: l/testing: 
ninja/1.11.1@local/testing: 
ninja/1.11.1@local/testing: Copying sources to build folder
ninja/1.11.1@local/testing: Building your package in /Users/regalir/.conan/data/ninja/1.11.1/local/testing/build/3eb32b976fb0626f3c40a8b87ace72156aa44204
ninja/1.11.1@local/testing: Generator txt created conanbuildinfo.txt
ninja/1.11.1@local/testing: Calling generate()
ninja/1.11.1@local/testing: Aggregating env generators
ninja/1.11.1@local/testing: Calling build()
ninja/1.11.1@local/testing: Package '3eb32b976fb0626f3c40a8b87ace72156aa44204' built
ninja/1.11.1@local/testing: Build folder /Users/regalir/.conan/data/ninja/1.11.1/local/testing/build/3eb32b976fb0626f3c40a8b87ace72156aa44204/build/Release
ninja/1.11.1@local/testing: Generated conaninfo.txt
ninja/1.11.1@local/testing: Generated conanbuildinfo.txt
ninja/1.11.1@local/testing: Generating the package
ninja/1.11.1@local/testing: Package folder /Users/regalir/.conan/data/ninja/1.11.1/local/testing/package/3eb32b976fb0626f3c40a8b87ace72156aa44204
ninja/1.11.1@local/testing: Calling package()
ninja/1.11.1@local/testing package(): WARN: No files in this package!
ninja/1.11.1@local/testing: Package '3eb32b976fb0626f3c40a8b87ace72156aa44204' created
ninja/1.11.1@local/testing: Created package revision 5f1337551a6b504be6c28c73fc6641cf
>>> command.run
(['-B', '-S', '-I', '-c', 'from multiprocessing.resource_tracker import main;main(13)'],)
'-B' is not a Conan command. See 'conan --help'.

ERROR: Unknown command '-B'
regalir@MTLRKJLYQV6G6 3dsmax-conan-recipes % 
``` 

We can now see that the first 'run' call receive valid conan arguments. But it is the second one that receive the -B option!
Note that I have absolutely no idea what step is run here but, hopefully it can help you narrow down what may be "wrong".

I also tried to install conan inside python with the command
` pip install conan==1.61`
and use it instead of my local build. So I ran:
`/Users/regalir/Library/Python/3.11/bin/conan create testing/ninja/conanfile.py 1.11.1@local/testing -pr=maya-2025-mac-release --build missing`

and this works without the -B error! Therefore, my conclusion is that conan have, somehow, a "dependency" on default stuff inside the site-packages that it cannot resolve if we have a stand-alone :-(

Hopefully, this long post gives you enough information to clarify what is going on and, maybe offer a viable solution?
Thanks!

**Comment by memsharded** [comment]
Thanks very much for the details @regalir , great investigation! 🙂 

Now this starts to make sense. It seems there is something very specific to the "compiled" or "bundled" Conan executable that is making it fail, and the standard Conan running from source or from ``pip`` doesn't have this issue.

Lets keep narrowing the scope! I don't have a Mac to test, so I'd need someone else from the team to help.
In the meantime there would be some extra things that would be very useful to check:

- Can we reproduce the problem with a standard recipe like the one created with ``conan new mypkg/1.0 -m=cmake_lib`` + ``conan create .`` without further arguments?
- It might be related to some changes in latests Python versions. It would be great if the same could be repeated with a previous 3.10 or 3.8 Python version
- Check and report the ``pyinstaller`` version, just in case

**Comment by regalir** [comment]
The "standard" recipe seems to be working fine. This is the output of the conan I built yesterday with my traces in it:

```
regalir@MTLRKJLYQV6G6 c % ../conan_src/pyinstaller/dist/conan/conan create .
>>> command.run
(['create', '.'],)
Exporting package recipe
mypkg/1.0 exports_sources: Copied 1 '.txt' file: CMakeLists.txt
mypkg/1.0 exports_sources: Copied 1 '.cpp' file: mypkg.cpp
mypkg/1.0 exports_sources: Copied 1 '.h' file: mypkg.h
mypkg/1.0: A new conanfile.py version was exported
mypkg/1.0: Folder: /Users/regalir/.conan/data/mypkg/1.0/_/_/export
mypkg/1.0: Using the exported files summary hash as the recipe revision: 340fd254093609f074c9fbd4fbaa89fe 
mypkg/1.0: Exported revision: 340fd254093609f074c9fbd4fbaa89fe
Configuration:
[settings]
arch=armv8
arch_build=armv8
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=15
os=Macos
os_build=Macos
[options]
[build_requires]
[env]

mypkg/1.0: Forced build from source
mypkg/1.0 (test package): Installing package
Requirements
    mypkg/1.0 from local cache - Cache
Packages
    mypkg/1.0:c6a7b978c36bfbdf05646cef650d3b8d3c8bea46 - Build

Installing (downloading, building) binaries...
mypkg/1.0: Configuring sources in /Users/regalir/.conan/data/mypkg/1.0/_/_/source/.
mypkg/1.0: Copying sources to build folder
mypkg/1.0: Building your package in /Users/regalir/.conan/data/mypkg/1.0/_/_/build/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46
mypkg/1.0: Generator txt created conanbuildinfo.txt
mypkg/1.0: Calling generate()
mypkg/1.0: WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
mypkg/1.0: Preset 'release' added to CMakePresets.json. Invoke it manually using 'cmake --preset release'
mypkg/1.0: If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE=/Users/regalir/.conan/data/mypkg/1.0/_/_/build/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46/build/Release/generators/conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=Release'
mypkg/1.0: Aggregating env generators
mypkg/1.0: Calling build()
mypkg/1.0: CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/Users/regalir/.conan/data/mypkg/1.0/_/_/build/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/regalir/.conan/data/mypkg/1.0/_/_/package/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/regalir/.conan/data/mypkg/1.0/_/_/build/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46/."
-- Using Conan toolchain: /Users/regalir/.conan/data/mypkg/1.0/_/_/build/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The CXX compiler identification is AppleClang 15.0.0.15000100
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done (0.3s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: /Users/regalir/.conan/data/mypkg/1.0/_/_/build/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46/build/Release
mypkg/1.0: CMake command: cmake --build "/Users/regalir/.conan/data/mypkg/1.0/_/_/build/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46/build/Release" '--' '-j20'
[ 50%] Building CXX object CMakeFiles/mypkg.dir/src/mypkg.cpp.o
[100%] Linking CXX static library libmypkg.a
[100%] Built target mypkg
mypkg/1.0: Package 'c6a7b978c36bfbdf05646cef650d3b8d3c8bea46' built
mypkg/1.0: Build folder /Users/regalir/.conan/data/mypkg/1.0/_/_/build/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46/build/Release
mypkg/1.0: Generated conaninfo.txt
mypkg/1.0: Generated conanbuildinfo.txt
mypkg/1.0: Generating the package
mypkg/1.0: Package folder /Users/regalir/.conan/data/mypkg/1.0/_/_/package/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46
mypkg/1.0: Calling package()
mypkg/1.0: CMake command: cmake --install "/Users/regalir/.conan/data/mypkg/1.0/_/_/build/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46/build/Release" --prefix "/Users/regalir/.conan/data/mypkg/1.0/_/_/package/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46"
-- Install configuration: "Release"
-- Installing: /Users/regalir/.conan/data/mypkg/1.0/_/_/package/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46/lib/libmypkg.a
-- Installing: /Users/regalir/.conan/data/mypkg/1.0/_/_/package/c6a7b978c36bfbdf05646cef650d3b8d3c8bea46/include/mypkg.h
mypkg/1.0 package(): Packaged 1 '.h' file: mypkg.h
mypkg/1.0 package(): Packaged 1 '.a' file: libmypkg.a
mypkg/1.0: Package 'c6a7b978c36bfbdf05646cef650d3b8d3c8bea46' created
mypkg/1.0: Created package revision b6ab568ea8a00b244ffe4d700f21b5c3
mypkg/1.0 (test package): Generator txt created conanbuildinfo.txt
mypkg/1.0 (test package): WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
mypkg/1.0 (test package): Generator 'CMakeDeps' calling 'generate()'
mypkg/1.0 (test package): WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
mypkg/1.0 (test package): Generator 'CMakeToolchain' calling 'generate()'
mypkg/1.0 (test package): Preset 'release' added to CMakePresets.json. Invoke it manually using 'cmake --preset release'
mypkg/1.0 (test package): If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE=/Users/regalir/Development/tests/c/test_package/build/Release/generators/conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=Release'
mypkg/1.0 (test package): Generator 'VirtualBuildEnv' calling 'generate()'
mypkg/1.0 (test package): Generator 'VirtualRunEnv' calling 'generate()'
mypkg/1.0 (test package): Aggregating env generators
mypkg/1.0 (test package): Generated conaninfo.txt
mypkg/1.0 (test package): Generated graphinfo
Using lockfile: '/Users/regalir/Development/tests/c/test_package/build/Release/generators/conan.lock'
Using cached profile from lockfile
mypkg/1.0 (test package): Calling build()
mypkg/1.0 (test package): CMake command: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/Users/regalir/Development/tests/c/test_package/build/Release/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/regalir/Development/tests/c/test_package/."
-- Using Conan toolchain: /Users/regalir/Development/tests/c/test_package/build/Release/generators/conan_toolchain.cmake
-- Conan: Target declared 'mypkg::mypkg'
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/regalir/Development/tests/c/test_package/build/Release
mypkg/1.0 (test package): CMake command: cmake --build "/Users/regalir/Development/tests/c/test_package/build/Release" '--' '-j20'
[ 50%] Linking CXX executable example
[100%] Built target example
mypkg/1.0 (test package): Running test()
mypkg/1.0: Hello World Release!
  mypkg/1.0: __aarch64__ defined
  mypkg/1.0: __cplusplus199711
  mypkg/1.0: __GNUC__4
  mypkg/1.0: __GNUC_MINOR__2
  mypkg/1.0: __clang_major__15
  mypkg/1.0: __apple_build_version__15000100

```

**Comment by franramirez688** [comment]
Hi @regalir 

I just opened a PR that seems to solve the issue https://github.com/conan-io/conan/pull/15628
Could you try the diff to check that it's working also on your side?

```diff
diff --git a/conans/conan.py b/conans/conan.py
index b67fa1ae0..561812e7c 100755
--- a/conans/conan.py
+++ b/conans/conan.py
@@ -1,5 +1,11 @@
 import sys
 import os
+import platform
+
+# https://github.com/icloud-photos-downloader/icloud_photos_downloader/commit/5450392a6d7c28c6926207b03e063f1ec049da2f#
+if platform.system() == "Darwin":
+    from multiprocessing import freeze_support
+    freeze_support() # fixing tqdm on macos

 if os.getenv("CONAN_V2_CLI"):
     from conans.cli.cli import main
```

**Comment by regalir** [comment]
Yes! These new lines removed the '-B' error :-)

So, simple question now, since 1.61 and 1.62 are already officially released, would this fix go into an 1.63 version? And, if so, how long would it take to become available? I can probably make a new package with my built local version but I think it would make much more sense that we use the official sources to build a new standalone binary.

**Comment by memsharded** [comment]
Closed by https://github.com/conan-io/conan/pull/15628, it will be in next 1.63 release. Thanks!

**Comment by franramirez688** [comment]
> So, simple question now, since 1.61 and 1.62 are already officially released, would this fix go into an 1.63 version? And, if so, how long would it take to become available?

Hi @regalir,
We're trying to release it this week, so it'll be available very soon 😁
